modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replicator/SchemaCompatibilityValidatorTest.java

```diff
@@ -300,11 +309,22 @@ private static void addInconvertible(ColumnType type1, ColumnType type2, List<Co
 
     @Test
     void combinationOfForwardCompatibleChangesIsCompatible() {
-        assertForwardCompatibleChangeAllowsCommitting(() -> List.of(
-                tableSchema(1, List.of(column(INT32, false))),
-                // Type is widened, NOT NULL dropped.
-                tableSchema(2, List.of(column(INT64, true)))
-        ));
+        assertForwardCompatibleChangeAllowsCommitting(new SchemaChangeSource() {
+            @Override
+            public List<FullTableSchema> schemaVersions() {
+                return List.of(
+                        tableSchema(1, List.of(column(INT32, false))),
+                        // Type is widened, NOT NULL dropped.
+                        tableSchema(2, List.of(column(INT64, true)))
+                );
+            }
+
+            @Override
+            // Not valid for compatible changes.
+            public List<String> expectedDetails() {
```

Comment:
How about adding `CompatibleSchemaChangeSource` defining this method in this way, so we don't need to repeat it everywhere?